mvrx/src/main/kotlin/com/airbnb/mvrx/mock/MvRxViewModelConfig.kt

```diff
@@ -0,0 +1,223 @@
+package com.airbnb.mvrx.mock
+
+
+import android.util.Log
+import com.airbnb.mvrx.BaseMvRxViewModel
+import com.airbnb.mvrx.MvRxState
+import com.airbnb.mvrx.MvRxStateStore
+import com.airbnb.mvrx.MvRxViewModelFactory
+import com.airbnb.mvrx.RealMvRxStateStore
+import java.util.LinkedList
+
+// TODO: Is there a better way to set this?
+var mvrxViewModelConfigProvider = MvRxViewModelConfigProvider()
```

Comment:
Apart from changing the debugMode boolean and changing properties, do you expect users to subclass the Provider? I realized that it is open.

Not sure how different it is but I would prefer keeping this Val and provide a function like below to configure further. 

```
mvrxConfigure(configure: MvRxViewModelConfigProvider.() -> Unit) {
   mvrxViewModelConfigProvider.configure()
}
```

Reply:
I don't think many people will need to subclass MvRxViewModelConfigProvider, but I left it open so that they could provide a custom implementation of `buildStateStore` if desired.

Any particular reasons you like the function configuration better? It seems like it would make it harder to override functions like `buildStateStore`, although we could have setter functions to override default behavior.

Needing to call a function like `mvrxConfigure` is a little less discoverable than just setting the val though

Reply:
I would do the function approach in case you have setters. It would look like a DSL and allow others to configure. I put that idea since some properties are variables already. 

I think if you want to make it open and provide people to replace the whole implementation, then shouldn't it be immutable?